TemporalAirlinesConcept.Services/Implementations/Flight/FlightWorkflow.cs

```diff
@@ -0,0 +1,245 @@
+    {
+        foreach (var ticket in _flight.Registered.Where(ticket =>
+                     ticket.PurchaseId == markTicketPaidSignalModel.PurchaseId))
+        {
+            ticket.PaymentStatus = PaymentStatus.Cancelled;
+        }
+
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Reserves a seat for a ticket in the flight.
+    /// </summary>
+    /// <param name="seatReservationSignalModel">The model containing the seat and ticket information.</param>
+    [WorkflowSignal]
+    public Task ReserveSeat(SeatReservationSignalModel seatReservationSignalModel)
+    {
+        var seat = _flight.Seats.FirstOrDefault(f => f.Name == seatReservationSignalModel.Seat);
+
+        if (seat is null)
+            throw new ApplicationFailureException($"Seat {seatReservationSignalModel.Seat} was not found.");
+
+        seat.TicketId = seatReservationSignalModel.TicketId;
+
+        var ticket = _flight.Registered.FirstOrDefault(t => t.Id == seatReservationSignalModel.TicketId);
+
+        if (ticket is null)
+            throw new ApplicationFailureException($"Ticket {seatReservationSignalModel.TicketId} was not found.");
+
+        ticket.Seat = seat.Name;
+
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Removes seat reservation.
+    /// </summary>
+    /// <param name="seatReservationSignalModel">The seat reservation request model that contains the seat information.</param>
+    [WorkflowSignal]
+    public Task ReserveSeatCompensation(SeatReservationSignalModel seatReservationSignalModel)
+    {
+        var registeredTicket = _flight.Registered.FirstOrDefault(t => t.Id == seatReservationSignalModel.TicketId);
+
+        if (registeredTicket is null)
+            return Task.CompletedTask;
+
+        registeredTicket.Seat = null;
+
+        var seat = _flight.Seats.FirstOrDefault(f => f.Name == seatReservationSignalModel.Seat);
+
+        if (seat is null)
+            return Task.CompletedTask;
+
+        seat.TicketId = null;
+
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Adds a passenger's ticket to the boarded list of a flight.
+    /// </summary>
+    /// <param name="boardingSignalModel">The model containing the information of the passenger's boarding request.</param>
+    [WorkflowSignal]
+    public Task BoardPassenger(BoardingSignalModel boardingSignalModel)
+    {
+        if (boardingSignalModel?.Ticket?.BoardingStatus is null)
+            return Task.CompletedTask;
+
+        boardingSignalModel.Ticket.BoardingStatus = BoardingStatus.Boarded;
+
+        _flight.Boarded.Add(boardingSignalModel.Ticket);
+
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Retrieves the flight details.
+    /// </summary>
+    /// <returns>The flight details.</returns>
+    [WorkflowQuery]
+    public FlightDetailsModel GetFlightDetails()
+    {
+        return _flight;
+    }
```

Comment:
```suggestion
    [WorkflowQuery]
    public FlightDetailsModel Flight { get; private set; }
```

Queries can be properties which makes sense for this use case